### PR TITLE
Feature development

### PR DESCRIPTION
**Pull request by Jeff-Thompson12**

### CODE REVIEW

**Comment by AARON-CLARK** [comment]
Wow @Jeff-Thompson12! I see this is still marked as a "draft", but I think it has plenty of improvements to justify a review (closing 5 open issues). I'd be happy to review before you add in any other features. I like reviewing the smallest possible PRs. It just makes it easier to wade through all the changes. Let me know if it's ready, thanks! Looks like it passed all the r checks.

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK Sure, that works for me.

File: R/mod_tableGen.R [hunk 1]
```diff
@@ -820,20 +830,9 @@ mod_tableGen_server <- function(input, output, session, datafile = reactive(NULL
...
-                  blockData$dataset), 
-          function(x,y,z,d) tidyCDISC::app_methods(x,y,z,
-                       group = {column() %quote% 'NULL'}, 
-                       data = tidyCDISC::data_to_use_str(d, ae_data, bds_data),
-                       totals = total_df)) %>%
-      map(setNames, tidyCDISC::common_rownames({Rscript_use_preferred_pop_data()}, {column() %quote% 'NULL'})) %>%
-      setNames(paste(blockData$gt_group)) %>%
-      bind_rows(.id = 'ID') %>%
-      mutate(ID = tidyCDISC::pretty_IDs(ID))
+      tg_table <- tidyCDISC::tg_gt(tg_datalist, blockData, total_df, {column() %quote% 'NULL'})
```
Comment (AARON-CLARK) [inline R/mod_tableGen.R:835]: Got the following error when running R script locally:

image

File: R/mod_tableGen.R [hunk 2]
```diff
@@ -820,20 +830,9 @@ mod_tableGen_server <- function(input, output, session, datafile = reactive(NULL
       # Calculate totals for population set
       {total_for_code()}
       
+      tg_datalist <- list(ADAE = ae_data, ADSL = bds_data, POPDAT = {Rscript_use_preferred_pop_data()})
```
Comment (AARON-CLARK) [inline R/mod_tableGen.R:833]: I loaded tidyCDISC without any ADAE data and got the following error: 

image

File: R/mod_tableGen_utils.R [hunk 3]
```diff
@@ -445,6 +445,38 @@ pretty_IDs <- function(ID) {
...
+    function(x,y,z,d) tidyCDISC::app_methods(x,y,z,
+                                             group = group, 
+                                             data = tidyCDISC::data_to_use_str(d, tg_datalist$ADAE, tg_datalist$ADSL),
+                                             totals = total_df)) %>%
+    purrr::map(setNames, tidyCDISC::common_rownames(tg_datalist$POPDAT, group)) %>%
+    setNames(paste(blockData$gt_group)) %>%
+    dplyr::bind_rows(.id = 'ID') %>%
+    dplyr::mutate(ID = tidyCDISC::pretty_IDs(ID))
+}
+
```
Comment (AARON-CLARK) [inline R/mod_tableGen_utils.R:479]: These new functions will need some `{roxygen2}` code to document and export where needed, plus will need some `{testthat}` tests!

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
Hi @Jeff-Thompson12, thanks for PR! I noticed a few issues - first, the Standard Analysis tables appear to be broken:

image

The console offers a little bit more info - stating the error came from an `if()` statement. I'm sure if you set `options(shiny.fullstacktrace = TRUE)`, it will reveal the source of the issue.

Couple other comments below!

FYI - this is another reason to keep PR's as small as possible, solving one issue at a time. half of the solutions look great, and half may not - but they'll all be hanging in limbo until everything is working. Admittedly, I guess we did solve a lot of R check issues in one big PR previously, but perhaps feature development reserves a little bit more honed focus for both developers and code reviewers benefit. I found this [small pull request manifesto](https://github.com/PlaytikaOSS/small-pull-request-manifesto) which outlines some benefits/ guidelines. Anyway, that's a side tangent! Thanks for the PR!

**Comment by Jeff-Thompson12** [comment]
> FYI - this is another reason to keep PR's as small as possible, solving one issue at a time. half of the solutions look great, and half may not - but they'll all be hanging in limbo until everything is working. Admittedly, I guess we did solve a lot of R check issues in one big PR previously, but perhaps feature development reserves a little bit more honed focus for both developers and code reviewers benefit. I found this [small pull request manifesto](https://github.com/PlaytikaOSS/small-pull-request-manifesto) which outlines some benefits/ guidelines. Anyway, that's a side tangent! Thanks for the PR!

Yes, I agree. And this PR could have been split into smaller parts. Maybe we could talk about naming conventions for PRs. That could help as well.

File: R/mod_tableGen_utils.R (hunk 3 above)
Comment (Jeff-Thompson12) [reply R/mod_tableGen_utils.R]: This is done in part. Examples and tests still need to be created, but appropriate functions have been exported.

File: R/mod_tableGen.R (hunk 2 above)
Comment (Jeff-Thompson12) [reply R/mod_tableGen.R]: I have updated the code to only include ADAE in the script if it was uploaded to the app.

**Comment by Jeff-Thompson12** [comment]
> Hi @Jeff-Thompson12, thanks for PR! I noticed a few issues - first, the Standard Analysis tables appear to be broken:
> 
> image
> 
> The console offers a little bit more info - stating the error came from an `if()` statement. I'm sure if you set `options(shiny.fullstacktrace = TRUE)`, it will reveal the source of the issue.

Problem identified and should be working now.

**Review by AARON-CLARK — APPROVED** [review]
lgtm

**Pull request merged**